waspc/src/Wasp/Generator/ServerGenerator/Common.hs

```diff
@@ -33,6 +35,12 @@ data ServerTemplatesDir
 
 data ServerTemplatesSrcDir
 
+class ValidServerTemplatesDir d
+
+instance ValidServerTemplatesDir ServerTemplatesDir
```

Comment:
**Nit:** Maybe a better name would be `ServerTemplatesLocation`. We basically have to make something up to avoid clashing with `ServerTemplatesDir` (which would obviously be the most appropriate name), but I think `Valid` is a bit weird (don't know how to argument why).

Reply:
> We basically have to make something up to avoid clashing with ServerTemplatesDir
Precisely :joy:

I like your suggestion, I'll take it.

Reply:
If we were to ditch the polymorphism, then this suggestion becomes no-op. Nice name though.